tensorflow/g3doc/hardware/adding_support/fake_device.md

```diff
@@ -0,0 +1,278 @@
+# Constructing a Fake Device in TensorFlow
+
+A good starting point to adding new hardware is to add 
+existing hardware under a different name! So here we add
+the standard CPU under the name SPU by constructing a 
+set of classes called ThreadPiscine (piscine means pool
```

Comment:
Based on my experience, I'm not sure it's going to be super helpful to show how to write another CPU device, given that custom hardware looks a lot different than CPUs in many ways.  In addition, there's a bunch of code that a custom device has to do that another CPU device wouldn't, so I think it might be better to be a bit more abstract at this point.